feat(kno-7174): add Expo SDK docs (#646)
knocklabs/docs: content/sdks/expo/push-notifications.mdx

---
title: "Handling Push Notifications with React Native and Expo"
description: A guide on integrating Expo push notifications with the Knock SDK in your React Native application.
section: SDKs
---

**Note:** This guide assumes you're utilizing our `KnockExpoPushNotificationProvider` for a streamlined push notification setup within your React Native and Expo environment.

## Prerequisites

Before diving into the integration process, ensure your Knock account is set up for push notifications. For initial setup instructions, please visit our [Push Notification Configuration Guide](https://docs.knock.app/integrations/push/overview).

> ## Step 1: Setting Up Push Notifications with Expo

Reviewer: This guide (and the RN one) might be a nice place to use our `<Step>` components ✨

1. **Create a Push Notification Channel in Knock:**

>    - Log in to your Knock dashboard and navigate to the Channels section.

Reviewer 2: ```suggestion
   - Log in to your Knock account and navigate to the **Integrations** > **Channels** section of the dashboard.
```
another small nit re location identification

   - Create a new channel with type `Expo Push Notifications` and note the channel ID.

2. **Install Expo Dependencies:**
   - `expo": "~50.0.14`
   - `expo-constants": "^15.4.0`
   - `expo-device": "^5.9.3`
   - `expo-notifications": "^0.27.6`

## Step 2: Registering for Push Notifications

1. **Wrap Your App with `KnockExpoPushNotificationProvider`:**

>    Ensure your app is wrapped with `KnockProvider` and then `KnockExpoPushNotificationProvider`, passing the Expo channel ID from Knock.

Reviewer 2: I think line 29 was meant to be a bullet point under this step (similar to #2) 

![Screenshot 2024-10-17 at 4 31 10 PM](https://github.com/user-attachments/assets/0de61e07-a6b0-4be0-9387-0ce562d9f857)

```jsx
import React from "react";
import { View } from "react-native";
import {
  KnockProvider,
  KnockExpoPushNotificationProvider,
} from "@knocklabs/expo";

export default function App() {
  return (
    <KnockProvider apiKey="{YOUR_KNOCK_PUBLIC_API_KEY}" userId="{YOUR_USER_ID}">
      <KnockExpoPushNotificationProvider knockExpoChannelId="{YOUR_KNOCK_EXPO_CHANNEL_ID}">
        <View> {/* Your app content here */} </View>
      </KnockExpoPushNotificationProvider>
    </KnockProvider>
  );
}
```

2. **Initiate Registration in Your Component:**

>    - The KnockExpoPushNotificationProvider automatically registers for push notifications.

Reviewer 2: ```suggestion
   - The `KnockExpoPushNotificationProvider` automatically registers for push notifications.
```

>    - The KnockExpoPushNotificationProvider automatically registers for push notifications.

Reviewer: ```suggestion
   - The `KnockExpoPushNotificationProvider` automatically registers for push notifications.
```

The `KnockExpoPushNotificationProvider` automatically handles receiving and tapping on notifications. To customize this behavior:

1. **Custom Notification Handling:**
   - Use the `onNotificationReceived` and `onNotificationTapped` methods from the `useExpoPushNotifications` hook to set custom handlers.

```jsx
useEffect(() => {
  onNotificationReceived((notification) => {
    console.log("Notification received:", notification);
  });

  onNotificationTapped((response) => {
    console.log("Notification tapped:", response);
  });
}, []);
```

## Step 4: Sending Test Notifications

Use the Knock dashboard or API to send a test notification to ensure your setup is correct. Verify that the notification appears on your device and that tapping on it triggers the expected behavior.

## Troubleshooting

- **Not Receiving Notifications:** Ensure your Expo push token is correctly registered with Knock and that your device's notification settings allow push notifications from your app.
- **Handling Silent Notifications:** If implementing silent notifications, ensure that your notification payload is correctly configured to not display an alert or sound.

> For further assistance, visit our [support documentation](https://docs.knock.app) or reach out to our support team.

Reviewer 2: Unsure if it makes sense to direct them to docs.knock.app when they are already at the docs site? We could update this to just have them reach out to our support team (and add a mailto:support@knock.app link)

> For further assistance, visit our [support documentation](https://docs.knock.app) or reach out to our support team.

Reviewer: ```suggestion
For further assistance, [reach out to our support team](mailto:support@knock.app).
```
I know this is a copy-paste from earlier content but I think maybe we should consider removing in both places? It feels meta to me to link to the docs from within the docs for further support 🤔
